frontend/src/components/dashboard/filters-section.tsx

```
@@ -48,174 +61,517 @@ export function FiltersSection({ data, loading, error, refetch }: FiltersSection
+    }
 
-      {/* Most Executed Tab */}
-      {activeTab === "most_executed" && (
-        <div className="overflow-x-auto">
-          {most_executed.length === 0 ? (
-            <div className="text-center py-8 text-muted-foreground">
-              No data available
+    // Sync tab selection when available tabs change (via effect to avoid render-loop).
+    const needsTabSync = availableTabs.length > 0 && !availableTabs.find(t => t.key === activeTab);
+    const syncTarget = needsTabSync ? availableTabs[0].key : null;
+    useEffect(() => {
```

Biome (2.3.14)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)